presto-main/src/main/java/com/facebook/presto/operator/GroupedTopNBuilder.java

```diff
@@ -69,19 +76,55 @@
     // keeps track sizes of input pages and heaps
     private long memorySizeInBytes;
     private int currentPageCount;
+    private LocalMemoryContext localUserMemoryContext;
+
+    public GroupedTopNBuilder(
+            OperatorContext operatorContext,
+            List<Type> sourceTypes,
+            List<Type> partitionTypes,
+            List<Integer> partitionChannels,
+            Optional<Integer> hashChannel,
+            int expectedPositions,
+            boolean isDictionaryAggregationEnabled,
+            JoinCompiler joinCompiler,
+            PageWithPositionComparator comparator,
+            int topN,
+            boolean produceRowNumber)
+    {
+        this(
+                operatorContext,
+                sourceTypes,
+                partitionTypes,
+                partitionChannels,
+                hashChannel,
+                expectedPositions,
+                isDictionaryAggregationEnabled,
+                joinCompiler,
+                comparator,
+                topN,
+                produceRowNumber,
+                UpdateMemory.NOOP);
+    }
 
     public GroupedTopNBuilder(
+            OperatorContext operatorContext,
             List<Type> sourceTypes,
+            List<Type> partitionTypes,
+            List<Integer> partitionChannels,
+            Optional<Integer> hashChannel,
+            int expectedPositions,
+            boolean isDictionaryAggregationEnabled,
+            JoinCompiler joinCompiler,
```

Comment:
It's anti pattern to pass operator-related fields into a non-operator class.... We probably don't need to change anything in this class. It's up to the calling operators to spill. `OrderByOperator` has the closest logic.

Reply:
Ah that's a good point. I borrowed this pattern from `SpillableHashAggregationBuilder` as I was implementing for TopNRowNumberOperator as well, which does grouped TopN and in that aspect `SpillableHashAggregationBuilder` is closer to this problem. I can reference `OrderByOperator` for code patterns.

Reply:
Update from offline discussions: 
Instead of moving `GroupByHash` into the GroupedTopNBuilder, we can create a GroupByHash Supplier/Factory and in the Operator pass it to the TopNBuilder.
This will avoid
1. leaking operator related fields (operatorContext, user/revocableMemoryContext, etc) and logic into TopNBuilder